pkg/agent/run.go

```diff
@@ -210,8 +211,20 @@ func gatherData(ctx context.Context, config Config) []*api.DataReading {
 			}
 			dgData, err := dg.Fetch()
 			if err != nil {
-				err = fmt.Errorf("%s: %v", k, err)
-				dgError = multierror.Append(dgError, err)
+				if _, ok := err.(*k8s.CRDNotFoundError); ok {
+					err := k8s.CRDNotFoundError{
+						Err:          err.Error(),
+						DataGatherer: k,
+					}
+					if StrictMode {
```

Comment:
@charlieegan3 @j-fuentes 

I'm not a fan of this part. I think the `getReadings()` function should not be responsible for handling the error edge cases. Ideally this would be handled by the backoff loop. But I haven't been able to find anything that could "ignore" a certain type of error in the [backoff package](https://pkg.go.dev/github.com/cenkalti/backoff/v4?tab=doc). 

Do you have any suggestions?

Reply:
I think that what you have looks ok - but as you say, it's a shame that we have know about the k8s error here in `Run`.

Perhaps datagathers can return an error or a permanent error. If they return a permanent error, then there will be no retries.

This way the dynamic dg can make a crd missing error a more generic 'PermanentError' signalling not to retry?

Reply:
I don't think using the permanent error would work. This would run into the problem of cancelling the retries if even if there are other errors that require a retry.

Reply:
> This would run into the problem of cancelling the retries if even if there are other errors that require a retry.

I don't understand this. If a data gatherer returns a `PermanentError`, then this should not continue processing that data gatherer and continue to the next (or fail hard if running with strict mode).